openfisca_core/tests/test_formulas_yaml_parameters.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+import os
+
+import numpy as np
+from nose.tools import assert_equal, assert_true
+from numpy.testing import assert_almost_equal
+
+from openfisca_core.taxbenefitsystems import MultipleXmlBasedTaxBenefitSystem
+
+from openfisca_core import parameters
+from openfisca_core.columns import IntCol, FloatCol
+from openfisca_core.formulas import Variable, set_input_divide_by_period, set_input_dispatch_by_period
+from openfisca_core.formula_helpers import switch
+from openfisca_core.tests import dummy_country
+from openfisca_core.tests.dummy_country import Individus
+
+
+class salaire_brut(Variable):
+    column = FloatCol
+    entity_class = Individus
+    label = "Salaire brut"
+    set_input = set_input_divide_by_period
+
+
+class plafond_securite_sociale(Variable):
+    column = FloatCol
+    entity_class = Individus
+    label = u"Plafond de la securite sociale"
+
+    def function(self, simulation, period):
+        instant = period.start.period(u'month').offset('first-of').start
+        # value = parameters.get(tax_benefit_system.parameters, 'parameters', 'plafond_securite_sociale', period.start)
+        value = self.get_parameter(instant)
+        return period, value
+
+
+class vieillesse_deplafonnee_salarie(Variable):
+    column = FloatCol
+    entity_class = Individus
+
+    def function(self, simulation, period):
+        instant = period.start.period(u'month').offset('first-of').start
+        pss = simulation.calculate('plafond_securite_sociale', period)
+        assiette = simulation.calculate('salaire_brut', period)
+
+        value = self.get_parameter(instant, base_options={'base': assiette, 'factor': pss})
```

Comment:
:+1: 
I forgot about this next step : introduce a sub class TemplateVariable where the arguments of get_parameter are given as class properties (or computed automatically)